pages/api/user.ts

```diff
@@ -1,46 +1,52 @@
 import { NextApiRequest, NextApiResponse } from 'next';
-import { prisma } from '../../lib/prisma';
+import prisma from '../../lib/prisma';
+import { User, Behavior } from '@prisma/client';
 
 export default async function handler(req: NextApiRequest, res: NextApiResponse) {
-  switch (req.method) {
+  switch ( req.method ) {
     case 'GET':
       return await getUser(req, res);
-    
-    /* case 'POST':
+  
+    case 'POST':
       return await createUser(req, res);
 
     case 'DELETE':
       return await deleteUser(req, res);
     
     case 'PUT':
-      return await updateUser(req, res); */
+      return await updateUser(req, res); 
 
     default:
       return res.status(405).end();
   }
 }
 
 const getUser = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { id } = req.query;
   try {
-    const user = await prisma.user.find({
+    const user = await prisma.user.findFirst({
       where: {
-        id: req.query.id
-      },
-      data: {
-        include: {
-          // include employee/patient info
-          EmployeeProfile: {
-            include: {
-              Patients: true
+        id: Number(id),
+        OR: [
+          {
+            StudentProfile: {
+              isNot: null
             }
           },
-          PatientProfile: true,
-        }
+          {
+            EmployeeProfile: {
+              isNot: null
+            }
+          }
+        ]
+      },
+      include: {
+        StudentProfile: true,
+        EmployeeProfile: true
       }
```

Comment:
So I may not have enough context, but I dont really understand why we need the OR here.
Also how come this is not in the search route instead? Are we trying to limit the complexity of the search route?

Reply:
All this stuff seems pretty unimplemented, might not be worth it to spend too much time on it right now.

Reply:
In my understanding, regarding the OR operator, this getUser function is supposed to fetch a single instance of user data, whether it belongs to a student or an employee, and specifically, it retrieves only the data of user whose profile data exists in the db.

And yes, the other functions, such as create/update/delete, weren't implemented in the original code before we had cloned it. But for now, I wanted to focus only on the part related to my issue, so I sort of left them untouched. In hindsight, I should have left it commented out for clarity purposes.

Reply:
Ok that makes more sense. I think maybe we can leave it as is for now and then come back to this file later.